runtime/lua/vim/_meta/builtin_types.lua

```diff
@@ -118,7 +118,7 @@
 --- @field topfill? integer
 --- @field topline? integer
 
---- @class vim.fn.winsaveview.ret
+--- @class vim.fn.winsaveview.ret: vim.fn.winrestview.dict
```

Comment:
I'm not sure about this change. Why inheritance? These two classes have the exactly same set of fields except that fields are optional/mandatory, so I don't see any point of inheriting from a base class. 

`vim.fn.winsaveview.ret` is a stricter type than `vim.fn.winrestview.dict`, so I feel like there should be no type error when passing a `vim.fn.winsaveview.dict` to `winrestview()`. Can you elaborate?

Reply:
Cause most lua LSPs don't view classes and tables the same: a class will never course into another class, even if they have the same field, unless the class is a parent of the other class.

If you want it to automatically course when fields match, then I'll recommend replacing the classes with tables in the format `--- @alias type_name {field: type}`.

Reply:
Oh you're right, I think this makes sense then.